File: android/chesslibjava/src/com/chesslib/board/PieceOnBoard.java
package com.chesslib.board;

import com.chesslib.Piece;
import com.chesslib.Player;

public final class PieceOnBoard {
    public final Player player;
    public final Piece piece;
    public final int boardSquare;

    public PieceOnBoard(Player player, Piece piece, int boardSquare) {
        this.player = player;
        this.piece = piece;
        this.boardSquare = boardSquare;
    }
}

File: ChessApp/ChessEngine/Board/PieceOnBoard.cs
namespace ChessEngine.Board
{
    public readonly struct PieceOnBoard
    {
        public readonly Player player;
        public readonly Piece piece;
        public readonly BoardSquare square;

        public PieceOnBoard(Player player, Piece piece, BoardSquare square)
        {
            this.player = player;
            this.piece = piece;
            this.square = square;
        }
    }
}
